### PR TITLE
Propagate Tag from PVC to PV

### LINKED ISSUE #66: Annotate PVs
**Is your feature request related to a problem? Please describe.**
It could be nice to have a way to tag PV with the tags provided

**Describe the solution you'd like**
- Add commandLine argument such as `--annotate-pv`
- If `--annotate-p` is enabled, pv will be annotated with the Tags
- If enabled it must be stated in the readme that SA will be able to `patch` `PersitentVolume`
https://github.com/mtougeron/k8s-pvc-tagger/blob/main/charts/k8s-pvc-tagger/templates/rbac.yaml#L73-L76

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

I can work on an PR if needed

### PR DESCRIPTION
**Pull request by speedfl**
Closes #66

Annotate PV from PVCs label

### CODE REVIEW

**Comment by mtougeron** [comment]
Awesome, I can't wait to review this! :)

**Comment by speedfl** [comment]
@mtougeron for information I am not a big fan of the name "propagate-to-pv" but I am not good finding names. So if you have an idea don't hesitate :D

File: README.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ The `k8s-pvc-tagger` watches for new PersistentVolumeClaims and when new AWS EBS
 
 `--allow-all-tags` - Allow all tags to be set via the PVC; even those used by the EBS/EFS controllers. Use with caution!
 
+`--propate-to-pv` - Allow to propagate labels and annotations from PVC to PV. Adding computed tags as annotation with prefix: `k8s-pvc-tagger/tags/`
```
Comment (mtougeron) [inline README.md:21]: What if we used `--add-metadata-to-pv`? Would that work? I'm horrible at naming too.

File: kubernetes.go [hunk 2]
```diff
@@ -169,13 +178,70 @@ func watchForPersistentVolumeClaims(ch chan struct{}, watchNamespace string) {
...
+		}
+	}
+
+	for key, value := range tags {
+		if isValidTagName(key) {
+			pv.ObjectMeta.Annotations[fmt.Sprintf(pvAnnotationTagFormat, key)] = value
+		}
+	}
+
+	newPV, err := k8sClient.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{})
```
Comment (mtougeron) [inline kubernetes.go:216]: Instead of `newPV` let's call this `updatedPV`

File: kubernetes.go [hunk 3]
```diff
@@ -343,26 +409,26 @@ func provisionedByAwsEbs(pvc *corev1.PersistentVolumeClaim) bool {
...
+func processPersistentVolumeClaim(pvc *corev1.PersistentVolumeClaim) (string, map[string]string, *corev1.PersistentVolume, error) {
 	tags := buildTags(pvc)
 
 	log.WithFields(log.Fields{"namespace": pvc.GetNamespace(), "pvc": pvc.GetName(), "tags": tags}).Debugln("PVC Tags")
 
 	pv, err := k8sClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
 	if err != nil {
 		log.WithFields(log.Fields{"namespace": pvc.GetNamespace(), "pvc": pvc.GetName()}).Errorln("Get PV from kubernetes cluster error:", err)
-		return "", nil, err
+		return "", nil, pv, err
```
Comment (mtougeron) [inline kubernetes.go:420]: If there's an error getting the PV then we should probably return nil for it.

File: kubernetes.go [hunk 4]
```diff
@@ -343,26 +409,26 @@ func provisionedByAwsEbs(pvc *corev1.PersistentVolumeClaim) bool {
...
 	}
 
 	var volumeID string
 	annotations := pvc.GetAnnotations()
 	if annotations == nil {
 		log.Errorf("cannot get PVC annotations")
-		return "", nil, errors.New("cannot get PVC annotations")
+		return "", nil, pv, errors.New("cannot get PVC annotations")
```
Comment (mtougeron) [inline kubernetes.go:427]: Let's be consistent and return nil for the PV in the event of errors like it does for the pvc

**Review by mtougeron — CHANGES_REQUESTED** [review]
I'm so sorry, I thought I had submitted this review weeks ago but it's been sitting here pending. Just some minor feedback.

**Pull request closed without merging**